packages/block-library/src/template-part/edit/advanced-controls.js

```diff
@@ -54,14 +55,17 @@ export function TemplatePartAdvancedControls( {
 		templatePartId
 	);
 
-	const definedAreas = useSelect( ( select ) => {
-		// FIXME: @wordpress/block-library should not depend on @wordpress/editor.
-		// Blocks can be loaded into a *non-post* block editor.
-		/* eslint-disable-next-line @wordpress/data-no-store-string-literals */
-		return select(
-			'core/editor'
-		).__experimentalGetDefaultTemplatePartAreas();
-	}, [] );
+	const defaultTemplatePartAreas = useSelect(
+		( select ) =>
+			select( coreStore ).getEntityRecord( 'root', '__unstableBase' )
+				?.default_template_part_areas || [],
+		[]
+	);
+
+	const definedAreas = defaultTemplatePartAreas.map( ( item ) => ( {
```

Comment:
Seems like we might not need this temporary variable and just move the mapping into the areaOptions `map`

Reply:
+1.

Reply:
Addressed with 7aab231e73075333b36b4d08727c012b9ca180a3